### PR TITLE
Fix for Android chip-tool build issue (#27284)

### LINKED ISSUE #27284: [Android arm64] Unable to build chip-tool - unknown argument: '-fmacro-prefix-map=../../='
### Reproduction steps

Issue Summary: 
Unable to build android arm64 for chip-tool through command line: './scripts/build/build_examples.py --target android-arm64-chip-tool build'

Issue Cause: clang++: error: unknown argument: '-fmacro-prefix-map=../../='

Please find below the error snippet.
2023-06-16 11:36:37 INFO    FAILED: obj/src/access/examples/libaccess.ExampleAccessControlDelegate.cpp.o 
2023-06-16 11:36:37 INFO    /home/test/Android/Sdk/ndk/21.4.7075529//toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android24-clang++ -MMD -MF obj/src/access/examples/libaccess.ExampleAccessControlDelegate.cpp.o.d -Wconversion -march=armv8-a -O0 -g2 -fno-common -ffunction-sections -fdata-sections -fvisibility=hidden -fno-exceptions -funwind-tables -fPIC -Wall -Werror -Wstack-usage=8192 -Wextra -Wshadow -Wunreachable-code -Wvla -Wformat -Wformat-nonliteral -Wformat-security -Wimplicit-fallthrough -Wheader-hygiene -Wshorten-64-to-32 -Wformat-type-confusion -Wthread-safety -Wno-deprecated-declarations -Wno-missing-field-initializers -Wno-unknown-warning-option -Wno-unused-parameter -fdiagnostics-color -fno-strict-aliasing -fmacro-prefix-map=../../= -std=gnu++14 -fno-rtti -Wnon-virtual-dtor -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS=1 -DCHIP_HAVE_CONFIG_H=1 -I../../src/include -I../../src -Igen/include -I../../zzz_generated/app-common -I../../config/standalone -I../../third_party/nlassert/repo/include -I../../third_party/nlio/repo/include -I../../third_party/nlfaultinjection/repo/include -I../../third_party/mbedtls/repo/include -c ../../src/access/examples/ExampleAccessControlDelegate.cpp -o obj/src/access/examples/libaccess.ExampleAccessControlDelegate.cpp.o
2023-06-16 11:36:37 INFO    clang++: error: unknown argument: '-fmacro-prefix-map=../../='

I am Sorry, I am unable to upload the build log file (it's not allowing me to upload when tried with error meassge: something went wrong, so I shared the error snippet above).

### Platform

android

### Platform Version(s)

_No response_

### Type

Platform validated

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by satyajitanand**
Fixes https://github.com/project-chip/connectedhomeip/issues/27284

This commit solves the above build issue of Android chip-tool project.

### CODE REVIEW

**Comment by andy31415** [comment]
@satyajitanand this seems to break linux builds - I believe these options were added to make error strings not contain developer-specific paths.

File: src/BUILD.gn [hunk 1]
```diff
@@ -30,7 +30,7 @@ declare_args() {
 
 # chip_root relative to root_build_dir for macro-prefix-map.
 # "/." avoids trailing "/" in result when chip_root is "//"
-build_relative_chip_root = rebase_path("${chip_root}/.", root_build_dir)
+# build_relative_chip_root = rebase_path("${chip_root}/.", root_build_dir)
```
Comment (andy31415) [inline src/BUILD.gn:33]: do not comment out things, remove them instead.

I believe these are intentional for non-android at least (and android should benefit from them too presumably).

If android really fails to build, then having a current_os check for android may work ... however our github CI passes, so I am unsure what is going on here.

**Review by andy31415 — CHANGES_REQUESTED** [review]
I believe this works in current CI. The original issue report is probably due to using an old NDK.

File: src/BUILD.gn [hunk 2]
```diff
@@ -45,7 +45,7 @@ config("includes") {
   defines = [ "CHIP_HAVE_CONFIG_H=1" ]
 
   # Make __FILE__ and related macros relative to chip_root
-  cflags = [ "-fmacro-prefix-map=${build_relative_chip_root}/=" ]
+  # cflags = [ "-fmacro-prefix-map=${build_relative_chip_root}/=" ]
```
Comment (bzbarsky-apple) [inline src/BUILD.gn:48]: This will expose full paths in error messages which is (1) a privacy leak and (2) larger code.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

**Comment by andy31415** [comment]
Closing: I don't think a code change is needed, a documentation change may be needed instead.

**Pull request closed without merging**